packages/test/test-utils/src/loaderContainerTracker.ts

```diff
@@ -209,6 +209,71 @@ export class LoaderContainerTracker implements IOpProcessingController {
         debugWait("Synchronized");
     }
 
+    /**
+     * Port of {@link @fluidframework/container-loader#waitContainerToCatchUp} for compatibility with old containers.
+     * Waits until container connects to delta storage and gets up-to-date
+     * Useful when resolving URIs and hitting 404, due to container being loaded from (stale) snapshot and not being
+     * up to date. Host may chose to wait in such case and retry resolving URI.
+     * Warning: Will wait infinitely for connection to establish if there is no connection.
+     * May result in deadlock if Container.disconnect() is called and never switched back to auto-reconnect.
+     * @returns true: container is up to date, it processed all the ops that were know at the time of first connection
+     *          false: storage does not provide indication of how far the client is. Container processed
+     *          all the ops known to it, but it maybe still behind.
+     * @throws an error if the container is closed before it catches up.
+     */
+    public async waitContainerToCatchUp(container: IContainer): Promise<boolean> {
```

Comment:
Another approach you might consider -- Could you monkey-patch on `connect` if it's missing, before calling `waitContainerToCatchUp`?

```
import { waitContainerToCatchUp as waitContainerToCatchUp_original } from ...;

...

const waitContainerToCatchUp = (container: IContainer) => {
  if (!(container as any).connect) {
    (container as any).connect = container.resume;
  }

  waitContainerToCatchUp_original(container);
}
```

(because this is yucky to have a copy... e.g. I'm changing this function in another PR now, and if I didn't see this PR I wouldn't know to update this too)

Reply:
@ChumpChief, what do you think about this solution? I think I prefer it to what we have now (duplicating the code).

Reply:
They both have their pros and cons I think - I'd personally be ok with either.

Reply:
@markfields, I implemented your solution. Let me know if you think it should be a private function, or even just define it as a const inside of `loadTestContainer()` like you showed in your example.